cc_core.c

```diff
@@ -65,6 +65,27 @@ void output_tokens(struct token_list *i, FILE* out)
 	}
 }
 
+int starts_with(char* str, char* needle)
```

Comment:
What is the advantage of `starts_with` rather than using `strncmcp`?

Reply:
I explicitly just want it to return a bool without having to remember what the return value means. I wouldn't mind using `strncmp` inside `starts_with` if you want?

Reply:
Sure we can do that (`strncmp` inside `starts_with`)

Reply:
We can't use `strncmp` since we don't know the length of `needle` without doing a `strlen` so we'll need to use `strcmp`. Am I dumb or is there no way to do it using only `strcmp`?

Reply:
Hmm, as long as strings are null terminated, strcmp should work just as well...

Reply:
Ah right, just check for `strcmp == 0`, right? The greater than/less than 0 return code made me confused.

Reply:
Pushed fix.